src/flash/flashCmd.ts

```diff
@@ -88,33 +104,128 @@ export async function verifyCanFlash(
+      const error = new Error(errorMessage);
+      const customButtons = [
+        {
+          label: "Change flash type to UART",
+          action: () => {
+            idfConf.writeParameter(
+              "idf.flashType",
+              "UART",
+              vscode.ConfigurationTarget.WorkspaceFolder,
+              workspaceRoot
+            );
+            const saveMessage = locDic.localize(
+              "flash.saveFlashTypeUART",
+              "Flashing method successfully changed to UART"
+            );
+            Logger.infoNotify(saveMessage);
+            OutputChannel.appendLineAndShow(saveMessage, "Flash Encryption");
+          },
+        },
+        {
+          label: "Disable Flash Encryption",
+          action: () => {
+            disableFlashEncryption();
+            const saveMessage = locDic.localize(
+              "menuconfig.disabledFlashEncryption",
+              "Flash encryption has been disabled in the SDK configuration"
+            );
+            Logger.infoNotify(saveMessage);
+            OutputChannel.appendLineAndShow(saveMessage, "Flash Encryption");
+          },
+        },
+      ];
+
+      OutputChannel.appendLineAndShow(errorMessage, "Flash Encryption");
+      Logger.errorNotify(errorMessage, error, { tag: "Flash Encryption" });
+      await showQuickPickWithCustomActions(
+        "Pick one of the following actions to continue",
+        customButtons
+      );
+      return {
+        success: false,
+        resultType: FlashCheckResultType.ErrorInvalidFlashType,
+      };
+    }
+    const eFuse = new ESPEFuseManager(workspaceRoot);
+    const summaryResult = await eFuse.readSummary();
+    if (summaryResult && summaryResult.BLOCK_KEY0) {
+      if (
+        // eFuse is not set
+        summaryResult.BLOCK_KEY0.value &&
+        summaryResult.BLOCK_KEY0.value === ESP.DEFAULT_UNSET_EFUSE_VALUE
+      ) {
+        //TODO: Flash normally and inform user it will be a normal flash and to reboot the device after flashing and flash again for encryption to take effect
+        const documentationUrl = await getDocsUrl(
+          ESP.URL.Docs.FLASH_ENCRYPTION,
+          workspaceRoot
+        );
+        const warnMessage =
+          "Encryption setup requires a two-step flashing process due to uninitialized eFuse BLOCK_KEY0. Please complete the current flash, reset your device, and then flash again. See documentation for details.";
+        showInfoNotificationWithLink(warnMessage, documentationUrl);
+        OutputChannel.appendLineAndShow(warnMessage, "Flash Encryption");
+        Logger.info(warnMessage, { tag: "Flash Encryption" });
+        return {
+          success: false,
+          resultType: FlashCheckResultType.ErrorEfuseNotSet,
+        };
+      }
+    }
+    return { success: true };
```

Comment:
This way the encrypt flashing will be executed even if the eFuse is not found, I think we should make sure that encryption is enabled in order to avoid accident encryption flashing

Reply:
Good catch! Fixed it.